extension/chrome/elements/compose.ts

```diff
@@ -64,43 +57,47 @@ Catch.try(async () => {
     }
     return await Store.passphraseGet(acctEmail, primaryKi.longid);
   };
-
-  await (async () => { // attempt to recover missing params
-    if (!isReplyBox || !threadId) {
-      return; // either not a reply box, or reply box & has all needed params
-    }
-    Xss.sanitizePrepend('#new_message', Ui.e('div', { id: 'loader', html: 'Loading secure reply box..' + Ui.spinner('green') }));
-    let gmailMsg;
-    try {
-      const thread = await Google.gmail.threadGet(acctEmail, threadId, 'metadata');
-      gmailMsg = await Google.gmail.msgGet(acctEmail, thread.messages[thread.messages.length - 1].id, 'metadata');
-    } catch (e) {
-      if (Api.err.isAuthPopupNeeded(e)) {
-        BrowserMsg.send.notificationShowAuthPopupNeeded(parentTabId, { acctEmail });
-      } else if (Api.err.isInsufficientPermission(e)) {
-        console.info(`skipping attempt to recover missing params due to insufficient permission`);
-      } else if (Api.err.isSignificant(e)) {
-        Catch.reportErr(e);
+  if (isReplyBox && threadId && !ignoreDraft && storage.drafts_reply && storage.drafts_reply[threadId]) {
+    draftId = storage.drafts_reply[threadId]; // there may be a draft we want to load
+  }
+  let from: string = '';
+  let subject: string = '';
+  let to: string[] = [];
+  let cc: string[] = [];
+  let bcc: string[] = [];
+  if (threadId) {
+    const fetchSuccess = await (async () => {
+      Xss.sanitizePrepend('#new_message', Ui.e('div', { id: 'loader', html: 'Loading secure reply box..' + Ui.spinner('green') }));
+      try {
+        const thread = await Google.gmail.threadGet(acctEmail, threadId, 'metadata');
+        const gmailMsg = await Google.gmail.msgGet(acctEmail, thread.messages[thread.messages.length - 1].id, 'metadata');
+        const reply = Google.determineReplyCorrespondents(acctEmail, storage.addresses || [], gmailMsg);
```

Comment:
If I'm reading this correctly, you could simplify as:
```js
const {from, to, cc, bcc} = Google.determineReplyCorrespondents(acctEmail, storage.addresses || [], gmailMsg);
```

Reply:
I couldn't make it exactly the same way you wrote but I have made it a little better